phys2bids/physio_obj.py

```diff
@@ -79,9 +79,93 @@ def has_size(var, data_size, token):
     return var
 
 
+def are_equal(self, other):
+    """
+    Return test of equality between two objects.
+
+    The equality is true if two objects are the same or
+    if one of the objects is equivalent to the dictionary
+    format of the other.
+
+    Parameters
+    ----------
+    other:
+        comparable object.
+
+    Returns
+    -------
+    boolean
+    """
+
+    def _deal_with_dict_value_error(self, other):
+        if self.keys() == other.keys():
+            alltrue_timeseries = [False] * len(self['timeseries'])
+            alltrue_keys = [False] * len(self)
+            for j, key in enumerate(self.keys()):
+                if key == 'timeseries':
+                    for i in range(len(self['timeseries'])):
+                        alltrue_timeseries[i] = (self['timeseries'][i].all()
+                                                 == other['timeseries'][i].all())
+                    alltrue_keys[j] = all(alltrue_timeseries)
+                else:
+                    alltrue_keys[j] = (self[key] == other[key])
+            return all(alltrue_keys)
+        else:
+            return False
+
+    try:
+        return self.__dict__ == other.__dict__
+    except ValueError:
+        try:
+            self.__dict__['timeseries']
+        except KeyError:
+            return False
+        except TypeError:
+            return False
+        else:
+            return _deal_with_dict_value_error(self.__dict__, other.__dict__)
+    except AttributeError:
+        try:
+            return self.__dict__ == other
+        except ValueError:
+            try:
+                self.__dict__['timeseries']
+            except KeyError:
+                return False
+            except TypeError:
+                return False
+            else:
+                return _deal_with_dict_value_error(self.__dict__, other)
+        except AttributeError:
+            try:
+                return self == other.__dict__
+            except ValueError:
+                try:
+                    self['timeseries']
+                except KeyError:
+                    return False
+                except TypeError:
+                    return False
+                else:
+                    return _deal_with_dict_value_error(self, other.__dict__)
+            except AttributeError:
+                try:
+                    return self == other
+                except ValueError:
+                    try:
+                        self['timeseries']
+                    except KeyError:
+                        return False
+                    except TypeError:
+                        return False
+                    else:
+                        return _deal_with_dict_value_error(self, other)
```

Comment:
Please write some comments. It's not easy to follow this code without any comments.